### PR TITLE
behaviors unnecessary for scanning removed from eth connection handler

### PR DESCRIPTION
**Pull request by SimonSK**
Removed most of the message handling logic
It should do the followings then end the connection:
- exchange Ethereum Status
- exchange GetBlockHeaders/BlockHeaders to verify if peers support DAO fork

Additionally, I hardcoded the DAO fork block header as the response to all GetBlockHeader requests in case the peer receives our information and decides to disconnect before we verify their info. ~The same info is used for our node's Ethereum Status, making it look like we are currently synced up to the DAO fork block (now I think about it, this is probably unnecessary given that the peers shouldn't be disconnecting us just because we aren't synced to a certain block. what do you think?)~ (reverted this change after discussion)

### CODE REVIEW

**Comment by zzma** [comment]
@SimonSK - I agree, I don't think we need to hardcode the BlockHeader messages. That shouldn't affect `HELLO` and `STATUS` messages

File: eth/peer.go [hunk 1]
```diff
@@ -230,10 +231,12 @@ func (p *peer) RequestReceipts(hashes []common.Hash) error {
 
 // Handshake executes the eth protocol handshake, negotiating version number,
 // network IDs, difficulties, head and genesis blocks.
-func (p *peer) Handshake(network uint64, td *big.Int, head common.Hash, genesis common.Hash) error {
+func (p *peer) Handshake(network uint64, chainconfig *params.ChainConfig, status *statusData) error {
```
Comment (zzma) [inline eth/peer.go]: Why did this change? And if this changed, are there no usages of `Handshake` that also need to be changed?

**Review by zzma — COMMENTED** [review]
One comment

File: eth/peer.go (hunk 1 above)
Comment (SimonSK) [reply eth/peer.go]: That change was to send a fake status message that says our node is synced up to the DAO fork block. As you said in the comment above, what we advertise in our Status message shouldn't affect peer connections, so I might revert this change. One benefit we get from hardcoding Status info, however, is that the client no longer needs to check its local blockchain state every time it sends the message.

**Comment by SimonSK** [comment]
@zzma Reverted the handshake behavior change back to the original

**Pull request merged**